### PR TITLE
[Flink-K8s-V2] Create Shaded module for flink-kubernetes-operator-api dependency

### LINKED ISSUE #2880: [Feature] [Flink-K8s-V2] Create shaded module for flink-kubernetes-operator-api dependency
It is a subtask of #2879 which includes the following content:
- Create a new module `streampark-shaded/streampark-shaded-flink-kubernetes-operator` with the purpose of isolating the references on which Flink Kubernetes Operator CR Bean definitions depend.
- The maven coordinates for shade are as follows:
  ```xml
  <dependency>
      <groupId>org.apache.flink</groupId>
      <artifactId>flink-kubernetes-operator-api</artifactId>
      <version>1.4.0</version>
  </dependency>
  ```

Design document: https://www.craft.me/s/LrTTzYHqSOLYqY

### PR DESCRIPTION
**Pull request by Al-assad**
## What changes were proposed in this pull request

Issue Number: close #2880 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): yes, including:
   -  org.apache.flink:flink-kubernetes-operator-api

### CODE REVIEW

**Comment by Al-assad** [comment]
@wolfboys PTAL

File: streampark-shaded/streampark-shaded-flink-kubernetes-operator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        <artifactId>streampark-shaded</artifactId>
+        <version>1.0.0</version>
+    </parent>
+
+    <artifactId>streampark-shaded-flink-kubernetes-operator-api</artifactId>
+    <name>StreamPark : Shaded Flink K8s Operator API</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
```
Comment (wolfboys) [inline streampark-shaded/streampark-shaded-flink-kubernetes-operator/pom.xml:32]: We need to declare the license in the [LICENSE](https://github.com/apache/incubator-streampark/blob/dev/dist-material/release-docs/LICENSE)

Comment (Al-assad) [reply streampark-shaded/streampark-shaded-flink-kubernetes-operator/pom.xml:32]: sure

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**